File: Golang/Medium/2221_Find_Triangular_Sum_of_an_Array/solution.go
package findtriangularsumofanarray

func triangularSum(nums []int) int {
	for len(nums) > 1 {
		tempArray := []int{}
		for i := 0; i < len(nums)-1; i++ {
			tempArray = append(tempArray, (nums[i]+nums[i+1])%10)
		}
		nums = tempArray
	}
	return nums[0]
}

File: Python/Medium/2221_Find_Triangular_Sum_of_an_Array/solution.py
class Solution:
    def triangularSum(self, nums: list[int]) -> int:

        while len(nums) > 1:
            tempArray = []
            for i in range(len(nums)-1):
                tempArray.append((nums[i] + nums[i+1]) % 10)
            
            nums = tempArray
        
        return nums[0]